Dockerfile

```diff
@@ -1,19 +1,18 @@
+# Build stage
 FROM golang:1.18 AS builder
 COPY . /build
 WORKDIR /build
 RUN CGO_ENABLED=0 go build ./carpenter.go
 
-
-FROM quay.io/centos/centos:stream8
-RUN dnf -y module install python39 &&\
-    dnf -y install python39 python39-pip &&\
-    python3.9 -m pip install --user --upgrade flake8
+# Main stage
+FROM python:3.9.16-alpine3.17
```

Comment:
From what I have read, while we get a smaller image and faster build time with `flake8` as our only python dependency in `python:alpine` that will [not always be the case](https://pythonspeed.com/articles/alpine-docker-python/). Moving to `python:alpine` opens us up to dealing with the complexities that can arise from Alpine building python packges from their source code because [standard PyPi wheels don't work on Alpine](https://pythonspeed.com/articles/alpine-docker-python/#standard-pypi-wheels-dont-work-on-alpine). To me, choosing `python:alpine` adds a significant tail risk of technical debt (like we won't see it right now, but it has the potential to be a huge headache later) for a minimal gain in build time and image size over a similar image like `python:slim-bullseye`. 

Ultimately, if we absolutely need the extra savings in image size, then I think we are locked into something like `python:alpine`.

Below is an extremely small sample size of image build times and sizes comparing `python:alpine` and `python:slim-bullseye`.



![python-alpine](https://user-images.githubusercontent.com/65633602/226689041-bd61f3b1-673b-4a51-9033-0b3bbda5c3ce.png)

Reply:
Fair enough. I think there is value in making this container as small as is reasonably possible due to the frequency it will run. Performance of the container is also important. Build time is less important since that should happen rarely.

After testing a few more base images, I've adjusted the dockerfiles to use the Debian bullseye slim image. This is roughly twice the size of the Alpine image, but still only 147MB, as you showed. I tried almalinux:8-minimal and centos:stream9-minimal, but both came in over 200MB. I think fedora minimal may come it a little smaller, but that also has potential maintenance issues, so the Debian build looks like the best compromise. right now